src/topp/PhosphoScoring.cpp

```diff
@@ -93,15 +92,150 @@ class TOPPPhosphoScoring :
 
 protected:
 
+  // spectrum must not contain 0 intensity peaks and must be sorted by m/z
+  template <typename SpectrumType>
+  static void deisotopeAndSingleChargeMSSpectrum_(SpectrumType& in, Int min_charge, Int max_charge, double fragment_tolerance, bool fragment_unit_ppm, bool keep_only_deisotoped = false, Size min_isopeaks = 3, Size max_isopeaks = 10, bool make_single_charged = true)
```

Comment:
Its actually a bit of code duplication but I wont find time till the release to factor this out of the search engine. So far it is just on my TODO list and I will pull it out when I find some time.